smoke/IotEdgeQuickstart/details/IotedgedWindows.cs

```diff
@@ -249,35 +251,38 @@ public async Task Stop()
                 // check service exists
                 if (iotedgeService != null)
                 {
-                    if (iotedgeService.Status == ServiceControllerStatus.Running)
+                    if (iotedgeService.Status != ServiceControllerStatus.Stopped)
                     {
                         iotedgeService.Stop();
-                        await Task.Delay(TimeSpan.FromSeconds(3));
+                        iotedgeService.WaitForStatus(ServiceControllerStatus.Stopped, TimeSpan.FromMinutes(2));
+                        iotedgeService.Refresh();
+
+                        if (iotedgeService.Status != ServiceControllerStatus.Stopped)
+                        {
+                            throw new Exception("Can't stop iotedge service within timeout period.");
+                        }
+
+                        // Add delay to ensure iotedge service is completely stopped.
+                        await Task.Delay(new TimeSpan(0, 0, 0, 5));
```

Comment:
Wait, iotedged is stopped but not completely stopped? What does that mean?

Reply:
I should remove it since I changed the stop method to use WaitForStatus instead.